### DISCUSSION TITLE
PSRules output into Power BI

### DISCUSSION THREAD

**Discussion by vishalgoyal16444** [microsoft/PSRule#987] [Show and tell]
Dear All,

We are running PSRules engine against our Azure subscription. 

1. 1st output we get is assets inventory as a JSON. We have ingested this into Power BI to build a report for easy review. Attached image shows the same
2.  Now we want to do same with PSRules output file. But that is not available as a JSON. Has someone done this. Any help would be appreciated.
![Picture1](https://user-images.githubusercontent.com/73927117/154920179-faf99d6d-9dd0-4f89-b371-73f6da34c591.png)

Thanks.

**Comment by BernieWhite** [comment]
@vishalgoyal16444 Very cool. Thanks for sharing. Interested in trying it myself now.

You can export PSRule result to JSON. It involves setting the [Output.Format](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#outputformat) option.

Also you will also want to set the [Output.Path](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#outputpath) option to configure the output path.

To get each instance set `Output.As` option to `Detail`.

It is also explained a little bit more here: https://microsoft.github.io/PSRule/v2/analysis-output/

You could save this to a storage account to pull into Power BI.

My example command line looks something like this `invoke-psrule -OutputFormat Json -OutputPath .\results.json -f . -Format File -as Detail`. But this could be used with `Assert-PSRule` or by setting options in `ps-rule.yaml`.

---

Separately you can also push results into Log Analytics and pull the data from Log Analytics into Power BI using the `PSRule.Monitor` module.

Info here: https://github.com/microsoft/PSRule.Monitor
Usage example here: https://azure.github.io/PSRule.Rules.Azure/setup/setup-azure-monitor-logs/

Information about how to pull this into Power BI is available here: https://docs.microsoft.com/azure/azure-monitor/logs/log-powerbi

Let us know how you go and if this information helps.

**Reply by vishalgoyal16444** [reply]
Thanks @BernieWhite. Appreciate quick detailed response. We will try the above and update this thread.

**Comment by vishalgoyal16444** [comment]
Thanks @BernieWhite. Your updates were helpful and we have progressed further and able to create the required Power BI report using JSON format. Appreciate your updates.
